discord/asset.py

```diff
@@ -209,11 +209,17 @@ def _from_cover_image(cls, state, object_id: int, cover_image_hash: str) -> Asse
 
     @classmethod
     def _from_guild_image(cls, state, guild_id: int, image: str, path: str) -> Asset:
+        animated = False
+        format = "png"
+        if path == "banners":
+            animated = image.startswith("a_")
+            format = "gif" if animated else "png"
+
         return cls(
             state,
-            url=f'{cls.BASE}/{path}/{guild_id}/{image}.png?size=1024',
+            url=f'{cls.BASE}/{path}/{guild_id}/{image}.{format}?size=1024',
```

Comment:
```suggestion
            url=f"{cls.BASE}/{path}/{guild_id}/{image}.{format}?size=1024",
```
Project code style fix